renderdoc/driver/gl/wrappers/gl_query_funcs.cpp

```diff
@@ -262,8 +262,12 @@ bool WrappedOpenGL::Serialise_glBeginQuery(GLenum target, GLuint qid)
 
   if(m_State < WRITING)
   {
-    m_Real.glBeginQuery(Target, GetResourceManager()->GetLiveResource(id).name);
-    m_ActiveQueries[QueryIdx(Target)][0] = true;
+    // Queries in the log interfere with the queries from FetchCounters.
+    if(!m_FetchCounters)
+    {
+      m_Real.glBeginQuery(Target, GetResourceManager()->GetLiveResource(id).name);
+      m_ActiveQueries[QueryIdx(Target)][0] = true;
```

Comment:
I think it's fine just to turn off the real queries for the duration of the counters run, as long as you do a full replay like I say to restore the correct state and texture contents.